Document/0x05b-Basic-Security_Testing.md

```diff
@@ -41,29 +41,50 @@ Note: On Linux, you need to choose an SDK directory. `/opt`, `/srv`, and `/usr/l
 
 #### Testing Device
 
+For dynamic analysis, you'll need an Android device to run the target app on. In principle, you can test without a real Android device and use only the emulator. However, apps execute quite slowly on the emulator, and this can make security testing tedious. Testing on a real device makes for a smoother process and a more realistic environment. On the other hand, emulators allow you to easily change SDK versions or create multiple devices. A full overview of the pros and cons of each approach is listed in the table below.
```

Comment:
This might be different with Correlium, genymotion and the virtual box based stack. So then we can compare emulator/simulator/real device

Reply:
Agree. With an emulator the performance issue is not really there anymore and is a differentiation to a simulator.

Reply:
Just to add a slight detail. some apps might never work properly on emulator owning to strict requirement of arm infra. most of the emulators be it genymotion, androidx86 or even android native emulator all now run x86 vm's so that becomes a pain point.